bioio_lif/reader.py

```diff
@@ -1,36 +1,59 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
-from typing import Any, Optional, Tuple
-
-from bioio_base.dimensions import Dimensions
-from bioio_base.reader import Reader as BaseReader
+import xml.etree.ElementTree as ET
+from copy import copy
+from math import floor
+from typing import Any, Dict, Hashable, List, Optional, Tuple, Union
 
+import dask.array as da
+import numpy as np
 import xarray as xr
+from bioio_base import constants, exceptions
+from bioio_base import io as io_utils
+from bioio_base import transforms, types
+from bioio_base.dimensions import (
+    DEFAULT_CHUNK_DIMS,
+    DEFAULT_DIMENSION_ORDER_LIST,
+    DEFAULT_DIMENSION_ORDER_LIST_WITH_MOSAIC_TILES,
+    REQUIRED_CHUNK_DIMS,
+    DimensionNames,
+    Dimensions,
+)
+from bioio_base.reader import Reader as BaseReader
+from dask import delayed
 from fsspec.spec import AbstractFileSystem
+from readlif.reader import LifFile
 
 ###############################################################################
 
 
 class Reader(BaseReader):
     """
-    The main class of each reader plugin. This class is subclass
-    of the abstract class reader (BaseReader) in bioio-base.
+    Wraps the readlif API to provide a bioio reader plugin for
+    volumetric LIF images.
 
     Parameters
     ----------
-    image: Any
-        Some type of object to read and follow the Reader specification.
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    chunk_dims: Union[str, List[str]]
+        Which dimensions to create chunks for.
+        Default: DEFAULT_CHUNK_DIMS
+        Note: Dimensions.SpatialY, Dimensions.SpatialX, and DimensionNames.Samples,
+        will always be added to the list if not present during dask array
+        construction.
```

Comment:
Can we add some comments here about the x and y params? Looks like I missed doing that when I originally added them. You can steal the comments from this snippet:
```
        # If either of these are True, the respective dimension will be
        # flipped along the other axis.
        # Ex. if `is_x_flipped = True` in a 4x4 tiled space, coordinate (2, 3)
        # will be (1, 3)
        # Ex. where both are true in a 4x4 tiled space, coordinate (1, 0)
        # will be (2, 3)
        # from the mosaic_position will be swapped such that field_x represents y
        # and field_y represents x.
        self.is_x_flipped = is_x_flipped
        self.is_y_flipped = is_y_flipped

        # If `is_x_and_y_swapped` is True, the field_x and field_y given
        # from the mosaic_position will be swapped such that field_x represents y
        # and field_y represents x.
```

Reply:
resolved in [c2d764a](https://github.com/bioio-devs/bioio-lif/pull/3/commits/c2d764abf04c346e342f56f798a1f32b5f20539c)